### PR TITLE
YJIT: Colorize outlined code differently on --yjit-dump-disasm

### PR DESCRIPTION
**Pull request by k0kubun**
On --yjit-dump-disasm, I'd like to make it easy to distinguish inline code and outlined code, which I feel makes it more readable. I made outlined code blue because it's a "cold" path, but I'm open to other ways of visualization.

### Before
Screenshot 2023-01-05 at 17 18 16

### After
Screenshot 2023-01-05 at 17 18 08

### CODE REVIEW

**Comment by noahgibbs** [comment]
I like this change. I don't *think* this output will ever get piped to a file. If it did, we might want to figure out how to disable it if not printing to terminal.

**Comment by k0kubun** [comment]
We already used `\x1b[1m` for comments, so not a new issue :slightly_smiling_face: Actually, I think it's more helpful to leave it in files too since you can see the colors in them with `less -R`.

File: yjit/src/disasm.rs [hunk 1]
```diff
@@ -175,13 +175,20 @@ pub fn disasm_addr_range(cb: &CodeBlock, start_addr: usize, end_addr: usize) ->
...
+
         // Comments for this block
         if let Some(comment_list) = cb.comments_at(insn.address() as usize) {
             for comment in comment_list {
-                writeln!(&mut out, "  \x1b[1m# {}\x1b[0m", comment).unwrap();
+                writeln!(&mut out, "  \x1b[1m{prefix}# {comment}{suffix}\x1b[0m").unwrap();
             }
         }
-        writeln!(&mut out, "  {}", insn).unwrap();
+        writeln!(&mut out, "  {prefix}{insn}{suffix}").unwrap();
```
Comment (maximecb) [inline yjit/src/disasm.rs:191]: Do you have to switch color for every single line or can you do it when the block begins and ends?

Comment (k0kubun) [reply yjit/src/disasm.rs]: I wasn't sure if you can disable bold while leaving blue, but it turns out it's possible with `\x1b[22m`. done https://github.com/ruby/ruby/pull/7073/commits/1b1f0f03dda19dff720c8d74baaca39f963fef8b

**Review by maximecb — APPROVED** [review]

**Review by XrXr — APPROVED** [review]
Looks nice!

File: yjit/src/disasm.rs (hunk 1 above)
Comment (noahgibbs) [reply yjit/src/disasm.rs]: I had wondered the same, but often it's better to start and end often. It means if output gets intermixed (e.g. with a background thread or process) then you won't wind up with accidental big areas of bold/coloured/standard/un-coloured when you didn't want them.

It's probably fine either way here.

**Pull request merged**